src/components/Image/Image.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import browserDimensions from '../../utils/browserDimensions';
+
+const Img = styled.img`
+  width: ${(props) => props.width};
+  height: ${(props) => props.height};
+  position: ${(props) => props.isLoaded ? 'static' : 'initial'};
+  visibility: ${(props) => props.src ? 'initial' : 'hidden'};
+  filter: ${(props) => props.grayscale ? 'grayscale(1)' : 'grayscale(0)'};
+  background-color: ${(props) => props.isLoaded ? 'transparent' : props.theme.color.translucent};
+`;
+
+class Image extends Component {
+  static getCdnUrl(src = '', aspectRatio) {
+    return src ? `${src}?w=${aspectRatio.width * 100}&h=${aspectRatio.height * 100}&fm=pjpg&fit=crop&auto=compress` : src;
+  }
+
+  state = {
+    src: typeof window !== 'undefined' && this.props.lazy ? '' : this.props.src,
```

Comment:
Initializing states with props should be avoided. The initialization of state from props only runs when the component is first created. The danger is that if the props on the component are changed without the component being ‘refreshed’, the new prop value will never be displayed because the constructor function will never update the current state of the component.